File: server/src/main/java/ehealth/data_objects/RegisterRequest.java
package ehealth.data_objects;

import com.fasterxml.jackson.annotation.JsonProperty;
import ehealth.enums.MedicalEffects;
import ehealth.enums.NegativeEffects;
import ehealth.enums.PositiveEffects;
import lombok.*;

import java.util.ArrayList;
import java.util.List;


@EqualsAndHashCode
@AllArgsConstructor
@Getter
@Setter
@NoArgsConstructor
@ToString
public class RegisterRequest {

    @JsonProperty("username")
    String username;

    @JsonProperty("password")
    String password;

    @JsonProperty("dob")
    String DOB;

    @JsonProperty("gender")
    String gender;

    @JsonProperty("country")
    String country;

    @JsonProperty("city")
    String city;

    @JsonProperty("medical")
    List<MedicalEffects> medical = new ArrayList<>();

    @JsonProperty("positive")
    List<PositiveEffects> positive = new ArrayList<>();

    @JsonProperty("negative")
    List<NegativeEffects> negative = new ArrayList<>();

}

File: client/CannaBe/CannaBe/DataObjects/Http/Requests/RegisterRequest.cs
using Newtonsoft.Json;
using System.Collections.Generic;

namespace CannaBe
{
    class RegisterRequest : LoginRequest
    {
        [JsonProperty("dob")]
        public string DOB { get; set; }

        [JsonProperty("gender")]
        public string Gender { get; set; }

        [JsonProperty("country")]
        public string Country { get; set; }

        [JsonProperty("city")]
        public string City { get; set; }

        [JsonProperty("medical")]
        public List<string> MedicalNeeds { get; set; }

        [JsonProperty("positive")]
        public List<string> PositivePreferences{ get; set; }

        [JsonProperty("negative")]
        public List<string> NegativePreferences { get; set; }

        public List<int> IntMedicalNeeds  { get; set; }
        public List<int> IntPositivePreferences  { get; set; }
        public List<int> IntNegativePreferences  { get; set; }

        public RegisterRequest() { }

        public RegisterRequest(string username, string password, string dob, string gender, string country, string city) 
            : base(username, password)
        {
            DOB = dob;
            Gender = gender;
            Country = country;
            City = city;
            MedicalNeeds = new List<string>();
            PositivePreferences = new List<string>();
            NegativePreferences = new List<string>();
        }
    }
}
